src/components/drag-and-drop/drag-and-drop.stories.js

```diff
@@ -0,0 +1,125 @@
+import React from 'react';
+import { storiesOf } from '@storybook/react';
+import { boolean } from '@storybook/addon-knobs';
+import { action } from '@storybook/addon-actions';
+import { State, Store } from '@sambego/storybook-state';
+import notes from './notes.md';
+import { DraggableContext } from './drag-and-drop';
+import {
+  Table,
+  TableHeader,
+  TableRow,
+  TableCell
+} from '../table';
+
+const store = new Store({
+  dndData: [{
+    id: '0',
+    name: 'UK'
+  }, {
+    id: '1',
+    name: 'Germany'
+  }, {
+    id: '2',
+    name: 'China'
+  }, {
+    id: '3',
+    name: 'US'
+  }]
+});
+
+const onDrag = (originalIndex, newIndex) => {
+  action('drag')();
+  const array = store.get('dndData');
+  const sortedItem = array.slice(originalIndex);
+  array.splice(originalIndex, 1);
+  array.splice(newIndex, 0, sortedItem[0]);
+  store.set({ dndData: array });
```

Comment:
I did it this way in `configurableItems`:
```
const handleDrag = (draggedIndex, hoveredIndex) => {
  const { configurableItemsData } = store.state;
  const draggedItem = configurableItemsData[draggedIndex];

  configurableItemsData[draggedIndex] = configurableItemsData[hoveredIndex];
  configurableItemsData[hoveredIndex] = draggedItem;

  store.set('configurableItemsData', configurableItemsData);
  action('dragged')();
};
```
Maybe could be helpful.

Reply:
I've made some changes here, I may have missed your point though. Let me know what you think.

Reply:
Little update, we should use  
`const configurableItemsData = store.get('configurableItemsData');`  
instead of  
`const { configurableItemsData } = store.state;`